AirCasting/SessionViews/SessionCartView.swift

```diff
@@ -57,7 +56,20 @@ struct SessionCartView: View {
         .onChange(of: session.allStreams) { _ in
             selectedStream = session.allStreams?.first
         }
-        .onAppear() {
+        .onChange(of: isFollowing) { _ in
+            let measurementStreamStorage = CoreDataMeasurementStreamStorage(persistenceController: persistenceController)
```

Comment:
1. If you really need to use storage here this should at least be injected and used as `MeasurementStreamStorage` protocol, not this concrete class. The protocol is designed so that clients don't know anything about coredata, persistence controllers etc. Use [injection](https://freecontent.manning.com/understanding-constructor-injection/) patterns

2. It would also be good if we didn't put databases inside views (at least with new code, like here) - there **really** should be some middle man objects in between. At least an orchestrating `ViewModel`, but I encourage more separation (ViewModel is still a view layer after all). Consider writing some tests for this - it looks like an important piece of logic.